### PR TITLE
fix(color): dirty storage flag to new source value assignment

### LINKED ISSUE #2465: Wigdet settings not stored prior to accepting USB connection
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

Updates if widget settings are lost after acceptable ng USB connection

### Expected Behavior

don't loose changes to model settings 

### Steps To Reproduce

can someone running RC2 on a color radio (TX16s) do me a favor and try this:
- On Screen setting select 2+1 widget and hit Setup widgets
- Click into one of the smaller rectangles and do Select widget
- Click on value and accept Rud as Source
- leave the page with RTN plus RTN to go to the main screen
- you should now see a widget returning rudder values
- long click on the widget and select widegt settings
- change Rud to Ele as source
- you should now see the widget returning elevator values
- now connect the radio to USB and accept the connection
- disconnect USB by ejecting the drive from the PC
- check the widget. Is Ele still the source? If not settings were lost

### Version

Nightly (Please give date/commit below)

### Transmitter

Radiomaster TX16S / TX16SMK2

### Anything else?

2.8.0-RC2

### PR DESCRIPTION
**Pull request by mha1**
Fixes #2465

Summary of changes: added dirty storage flag to new source value assignment

### CODE REVIEW

**Comment by gagarinlg** [comment]
> @pfeerick This fixes the settings widget pop up issue where the source value is not forced written to .yml. I wonder if this shouldn't be added to all switch (option.type) cases as all of them potentially change values. Please let me know, will extend if you think this is the right ting to do.

sounds right to me

**Comment by mha1** [comment]
@gagarinlg I've added a SET_DIRTY to all cases

File: radio/src/gui/colorlcd/widget_settings.cpp [hunk 1]
```diff
@@ -101,6 +102,7 @@ WidgetSettings::WidgetSettings(Window* parent, Widget* widget) :
         new ModelTextEdit(line, rect_t{},
                           widget->getOptionValue(optIdx)->stringValue,
                           sizeof(widget->getOptionValue(optIdx)->stringValue));
+        SET_DIRTY();
```
Comment (gagarinlg) [inline radio/src/gui/colorlcd/widget_settings.cpp:105]: This does not work here. With your change, the widget will always be marked as dirty.
The ModelTextEdit takes care of marking the model data dirty when the text is changed by itself. You can just remove this here.

File: radio/src/gui/colorlcd/widget_settings.cpp [hunk 2]
```diff
@@ -147,6 +150,7 @@ WidgetSettings::WidgetSettings(Window* parent, Widget* widget) :
             [=](int16_t newValue) {  // setValue
               widget->getOptionValue(optIdx)->unsignedValue =
                   (uint32_t)newValue;
+            SET_DIRTY();
```
Comment (gagarinlg) [inline radio/src/gui/colorlcd/widget_settings.cpp:153]: can you please check the indenttaion for all changes?
The SET_DIRTY call should be on the same level as "widget->"

**Review by gagarinlg — CHANGES_REQUESTED** [review]

File: radio/src/gui/colorlcd/widget_settings.cpp (hunk 2 above)
Comment (mha1) [reply radio/src/gui/colorlcd/widget_settings.cpp]: @gagarinlg Thanks for your review Malte. Changed code accordingly.

**Comment by mha1** [comment]
@gagarinlg  please check my last commit. should be ok now.

**Review by gagarinlg — APPROVED** [review]

**Comment by gagarinlg** [comment]
perfect, thx

**Pull request merged**